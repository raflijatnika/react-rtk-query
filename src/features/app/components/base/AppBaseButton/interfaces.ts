// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
