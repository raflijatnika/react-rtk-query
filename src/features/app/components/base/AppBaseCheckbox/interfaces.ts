// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
