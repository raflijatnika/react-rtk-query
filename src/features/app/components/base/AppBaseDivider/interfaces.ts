// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
