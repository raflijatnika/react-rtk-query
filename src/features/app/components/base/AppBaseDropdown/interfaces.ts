// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
