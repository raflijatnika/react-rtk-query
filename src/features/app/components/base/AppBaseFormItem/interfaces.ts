// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
