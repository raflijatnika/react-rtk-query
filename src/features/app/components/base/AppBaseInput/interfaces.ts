// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
