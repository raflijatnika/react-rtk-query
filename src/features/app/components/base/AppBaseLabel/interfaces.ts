// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
