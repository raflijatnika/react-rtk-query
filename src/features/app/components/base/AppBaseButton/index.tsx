// Styled Components
import styled from 'styled-components'

// Antd
import { Button } from 'antd'

// Interfaces
import { IAppBaseButtonProps } from './interfaces'

// Lodash
import omit from 'lodash.omit'

const AppBaseButton = styled((props: IAppBaseButtonProps) => (
  <Button
    {...omit<IAppBaseButtonProps>(props, [
      'backgroundColor',
      'fontColor',
      'width',
      'height'
    ])}
  />
))`
  background-color: ${props =>
    !props?.type && props?.backgroundColor
      ? props?.backgroundColor
      : undefined};
  color: ${props =>
    !props?.type && props?.fontColor ? props?.fontColor : undefined};
  width: ${props => (props?.width ? `${props.width}px` : 'auto')};
  height: ${props => (props?.height ? `${props.height}px` : 'auto')};
  border-radius: 10px;
  font-weight: 500px;
  font-size: 16px;
  border: none;
  line-height: 21px;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
`

export { AppBaseButton }
