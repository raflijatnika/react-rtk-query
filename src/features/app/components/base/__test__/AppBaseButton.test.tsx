// Test
import { render } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseButton } from '../AppBaseButton'

// Interfaces
import { IAppBaseButtonProps } from '../AppBaseButton/interfaces'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

const rendererInstance = (props?: IAppBaseButtonProps): ReactTestRenderer =>
  renderer.create(<AppBaseButton {...props} />)

it('AppBaseButton should render', () => {
  render(<AppBaseButton />)
})

it('AppBaseButton should have snapshot', () => {
  const tree = rendererInstance().toJSON()
  expect(tree).toMatchSnapshot()
})

it('AppBaseButton should have default style', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toHaveStyleRule('border-radius', '10px')
  expect(tree).toHaveStyleRule('font-weight', '500px')
  expect(tree).toHaveStyleRule('font-size', '16px')
  expect(tree).toHaveStyleRule('border', 'none')
  expect(tree).toHaveStyleRule('line-height', '21px')
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('align-items', 'center')
  expect(tree).toHaveStyleRule('justify-content', 'center')
  expect(tree).toHaveStyleRule('gap', '8px')
})

it('AppBaseButton should render correct props', () => {
  const tree = rendererInstance({
    backgroundColor: APP_COLOR_LIGHT.BACKGROUND,
    fontColor: APP_COLOR.DARK,
    width: 300,
    height: 500
  }).toJSON()

  expect(tree).toHaveStyleRule('background-color', APP_COLOR_LIGHT.BACKGROUND)
  expect(tree).toHaveStyleRule('color', APP_COLOR.DARK)
  expect(tree).toHaveStyleRule('width', '300px')
  expect(tree).toHaveStyleRule('height', '500px')
})
