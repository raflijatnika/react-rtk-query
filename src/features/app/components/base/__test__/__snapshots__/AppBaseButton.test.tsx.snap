// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseButton should have snapshot 1`] = `
.c0 {
  width: auto;
  height: auto;
  border-radius: 10px;
  font-weight: 500px;
  font-size: 16px;
  border: none;
  line-height: 21px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 8px;
}

<button
  className="ant-btn ant-btn-default c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
